tests/test_wog.py

```diff
@@ -114,6 +114,23 @@ def test_chogall_cannot_pay_health():
 	assert not fireball.is_playable()
 
 
+def test_demented_frostcaller():
+	game = prepare_game()
+	game.player1.give("OG_085").play()
+	game.end_turn()
+	# Fill enemy board with wisps.
+	for _ in range(7):
+		game.player2.give(WISP).play()
+	game.end_turn()
+	enemies = game.player2.characters
+	# Cast 8 coins, ensuring enemies are frozen one by one.
+	for x in range(1,9):
+		game.player1.give(THE_COIN).play()
+		assert len([en for en in enemies if en.frozen]) == x
+	# Cast one extra coin, ensuring nothing breaks when all enemies are already frozen.
+	game.player1.give(THE_COIN).play()
+
```

Comment:
You don't need such an extensive test. A single wisp with 3 coin casts (hero, wisp and null) is fine.

Also remember to add newlines after `end_turn()`.